src/lwip/freertos/sys_arch.c

```diff
@@ -274,12 +274,11 @@ u32_t sys_now(void)
 
 sys_prot_t sys_arch_protect(void)
 {
-    taskENTER_CRITICAL();
-    return 1;
+    return taskENTER_CRITICAL_FROM_ISR();
```

Comment:
Ah, so this is callable even outside of ISR context?

Reply:
Yup.  On NRF52, both variants of `ENTER_CRITICAL` do the same thing -- they both ultimately call `vPortRaiseBASEPRI()`.  The difference is that on exit, the task-variant restores the interrupts to 0, whereas the ISR variant restores them to a saved value.  There is a more significant difference on the nRF51 series --  in ISR variant, the code disables interrupts, in task variant with softdevice present, the code calls `sd_nvic_critical_region_enter` which presumably permits some sd-related ISRs to proceed unmolested.